"o arquivo usava ";" entre colunas e "," nos valores decimais" 
aparecia mensagem de erro parsererror por causa disso

solucao usar SEP

In [19]:
import pandas as pd
import numpy as np

Real_State = pd.read_csv("Real estate valuation data set.csv", sep=";", decimal=",")
Real_State.shape


(414, 8)

arquivo com 414 linhas e 8 colunas

In [20]:
Real_State.head()

,No,X1 transaction date,X2 house age,X3 distance to the nearest MRT station,X4 number of convenience stores,X5 latitude,X6 longitude,Y house price of unit area
0,1,2012.917,32.0,84.87882,10,24.98298,121.54024,37.9
1,2,2012.917,19.5,306.59470,9,24.98034,121.53951,42.2
2,3,2013.583,13.3,561.98450,5,24.98746,121.54391,47.3
3,4,2013.500,13.3,561.98450,5,24.98746,121.54391,54.8
4,5,2012.833,5.0,390.56840,5,24.97937,121.54245,43.1


"Informações Adicionais sobre Variáveis
As entradas são as seguintes
X1=a data da transação (por exemplo, 2013.250=março de 2013, 2013.500=junho de 2013, etc.)
X2=a idade da casa (unidade: ano)
X3=a distância até a estação de MRT mais próxima (unidade: medidor)
X4=o número de lojas de conveniência no círculo de vida a pé (inteiro)
X5=a coordenada geográfica, latitude. (unidade: grau)
X6=a coordenada geográfica, longitude. (unidade: grau)

A saída é a seguinte
Y = preço da casa por área unitária (10.000 Novo Dólar de Taiwan/Ping, onde Ping é uma unidade local, 1 Ping = 3,3 metros quadrados) "


como X1 nao esta em um formato correto tenho que realizar a transformação dos dados 
coverter, a ideia é:

2013 = ano
0.250 = 3 meses do ano
0.500 = 6 meses do ano
0.750 = 9 meses do ano

In [21]:
# Corrige o nome da coluna e converte X1 para mês-ano
Real_State = Real_State.rename(columns={"X1 transaction date": "X1"})

Real_State["X1"] = pd.to_numeric(Real_State["X1"], errors="coerce")
Real_State["ano"] = Real_State["X1"].astype(int)
Real_State["fracao_ano"] = Real_State["X1"] - Real_State["ano"]
Real_State["mes"] = (Real_State["fracao_ano"] * 12).round().astype(int)
Real_State["mes"] = Real_State["mes"].clip(lower=1, upper=12)
Real_State["mes_ano"] = Real_State["mes"].apply(lambda m: f"{m:02d}") + "-" + Real_State["ano"].astype(str)

Real_State[["X1", "mes_ano"]].head()

Real_State.head()

,No,X1,X2 house age,X3 distance to the nearest MRT station,X4 number of convenience stores,X5 latitude,X6 longitude,Y house price of unit area,ano,fracao_ano,mes,mes_ano
0,1,2012.917,32.0,84.87882,10,24.98298,121.54024,37.9,2012,0.917,11,11-2012
1,2,2012.917,19.5,306.59470,9,24.98034,121.53951,42.2,2012,0.917,11,11-2012
2,3,2013.583,13.3,561.98450,5,24.98746,121.54391,47.3,2013,0.583,7,07-2013
3,4,2013.500,13.3,561.98450,5,24.98746,121.54391,54.8,2013,0.500,6,06-2013
4,5,2012.833,5.0,390.56840,5,24.97937,121.54245,43.1,2012,0.833,10,10-2012
